_posts/2023-10-03-securing-wildfly-apps-oidc-openshift.adoc

```diff
@@ -0,0 +1,346 @@
+
+. Add a new user named *alice*. Set an *email* address for this new user, we'll use *alice@example.org*. For more information, see the Keycloak documentation on how to https://www.keycloak.org/getting-started/getting-started-openshift#_create_a_user[create a user].
+
+. Once the new user has been created, set a password for this new user from the *Credentials* tab.
+
+. From the *Role Mapping* tab, assign *alice* the *user* role. For more information, see the Keycloak documentation on how to https://www.keycloak.org/docs/latest/server_admin/index.html#proc-assigning-role-mappings_server_administration_guide[assign a role] to a user.
+
+. Create a new client as follows:
+* *General Settings*:
+** *Client type* (or *Client Protocol*, depending on your Keycloak version): *openid-connect*
+** *Client ID*: *simple-webapp-oidc*
+* *Capability config*:
+** *Authentication flow*: *Standard flow*, *Direct access grants*
+* *Login settings*: Leave the fields blank for now.
+
+For more information, see the Keycloak documentation on how to https://www.keycloak.org/docs/latest/server_admin/index.html#_oidc_clients[Managing OpenID Connect clients].
+
+. Click *Action*, *Download adapter config*, and select `Keycloak OIDC JSON` as the *Format Option* to see the connection parameters.
+
+== Update the example application and push it to your fork
+
+. Navigate to the `src/main/webapp/WEB-INF` directory in the application.
+
+. Update the values in the `oidc.json` file as follows:
++
+[source,json]
+----
+{
+    "client-id" : "myclient",                                    <1>
+    "provider-url" : "${env.OIDC_PROVIDER_URL:http://localhost:8080}/realms/myrealm", <2>
+    "public-client" : "true",
+    "principal-attribute" : "preferred_username",
+    "ssl-required" : "EXTERNAL"
+}
+----
+<1> Change to the value of `client-id` defined in Keycloak: `simple-webapp-oidc`.
+<2> Change to the value of `provider-url` to the Keycloak realm URL. Use this command to obtain the URL:
++
+[source]
+----
+KEYCLOAK_URL=https://$(oc get route keycloak --template='{{ .spec.host }}') && echo "" && echo "Keycloak myrealm URL:   $KEYCLOAK_URL/realms/myrealm" && echo ""
+----
++
+WildFly's native OIDC interface uses the `oidc.json` configuration file to configure OIDC.
+
+. Push the changes to your fork.
++
+[source]
+----
+cd /PATH/TO/ELYTRON/EXAMPLES/simple-webapp-oidc
+git add src/main/webapp/WEB-INF/oidc.json
+git commit -m "Added OIDC configuration file"
+git push origin HEAD
+---- 
+
+== Add Helm Configuration
+
+Let's switch to the `charts` directory in our `simple-webapp-oidc` example:
+
+[source]
+----
+cd /PATH/TO/ELYTRON/EXAMPLES/simple-webapp-oidc/charts
+----
+
+Notice there's a `helm.yaml` file in this directory with the following content:
+
+[source]
+----
+build:
+  uri: https://github.com/YOUR_GITHUB_USERNAME/elytron-examples.git <1>
+  contextDir: simple-webapp-oidc
+  ref: YOUR_BRANCH                                                  <2>
+deploy:
+  replicas: 1
```

Comment:
I think this is where we could potentially add an env section, e.g.,

```
deploy:
     env:
        -name: OIDC_PROVIDER_URL
        -value: 
```

Reply:
thanks! done